### PR TITLE
Add comprehensive E2E tests for onboarding and progressive disclosure

### PR DESCRIPTION
**Pull request by IsmaelMartinez**
## Summary
This PR adds comprehensive end-to-end test coverage for the onboarding wizard and progressive disclosure features using Playwright. These tests ensure that new users are properly guided through the application and that features unlock based on user engagement metrics.

### Onboarding Wizard Tests (`tests/onboarding.spec.ts`)
- **Display tests**: Verify wizard appears for new users (setupCompleted: false) and is hidden for returning users
- **Path navigation tests**: Validate all three onboarding paths (Explore, Plan, Ask) navigate correctly and show appropriate guidance content
- **Skip functionality**: Confirm "Skip for now" closes the wizard and prevents re-appearance
- **Back navigation**: Test ability to return to previous screens and select different paths
- **Completion state**: Verify setupCompleted flag is set after completing the flow and wizard doesn't reappear
- **Mobile responsiveness**: Ensure wizard displays and functions correctly on mobile viewports (375x667)

### Progressive Disclosure Tests (`tests/progressive-disclosure.spec.ts`)
- **Navigation tests**: Verify new users see only 3 primary nav items (Today, This Month, Seeds) with locked features in a "More" dropdown
- **AI Advisor unlock**: Test unlocking after 3 visits OR 1 planting
- **Compost unlock**: Test unlocking after 5 visits OR first harvest
- **Allotment Layout unlock**: Test unlocking after 5 plantings
- **Manual unlock**: Verify "Unlock now" buttons manually unlock features and persist across page reloads
- **Progress indicators**: Confirm locked features show progress bars and unlock thresholds (e.g., "4/5")
- **Mobile support**: Test progressive disclosure works correctly on mobile with menu expansion

## Implementation Details

- **Helper functions**: Reusable data factories for creating new user data, existing user data, and allotment data with various states (plantings, harvests)
- **localStorage setup**: Tests use `addInitScript` to pre-populate localStorage with test data before navigation
- **Engagement tracking**: Tests verify the `allotment-engagement` object tracks visits and manual unlocks
- **Accessibility**: Tests use semantic queries (getByRole, getByText) for better maintainability
- **Viewport testing**: Separate mobile test suites with 375x667 viewport to ensure responsive design

## Test Coverage
- 476 lines: 13 test suites, 30+ individual tests for onboarding
- 520 lines: 8 test suites, 20+ individual tests for progressive disclosure
- Covers happy paths, edge cases, persistence, and mobile scenarios

https://claude.ai/code/session_015ahfVdgjhfo77dDJnweznj

### CODE REVIEW

File: tests/progressive-disclosure.spec.ts [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+  await page.addInitScript(() => {
+    localStorage.setItem('allotment-unified-data', JSON.stringify({
+      meta: { setupCompleted: true },
+      layout: { areas: [] },
+      seasons: [],
+      currentYear: new Date().getFullYear(),
+      varieties: []
+    }))
+  })
+}
```
[the remark this answers is not in the document]
Comment (IsmaelMartinez) [reply tests/progressive-disclosure.spec.ts]: Acknowledged. This is a valid refactoring suggestion for a future PR. The duplication is intentional in test files to keep tests self-contained and reduce cross-file dependencies during test development.

File: tests/progressive-disclosure.spec.ts [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+  test('locked features show lock icon and progress bar in More dropdown', async ({ page }) => {
+    await skipOnboarding(page)
+    await page.goto('/')
+
+    // Open More dropdown
+    const moreButton = page.locator('header button').filter({ hasText: 'More' })
+    await moreButton.click()
+
+    // Check for lock icons (locked features show Lock icon overlay)
+    const lockIcons = page.locator('[role="menu"]').locator('svg').filter({ has: page.locator('path[d*="M16"]') })
```
[the remark this answers is not in the document]
Comment (IsmaelMartinez) [reply tests/progressive-disclosure.spec.ts]: Fixed in commit 7ea05e9 - replaced the fragile SVG selector with a semantic check for 'Unlock now' buttons.
